JavaFXUserInterface/src/main/java/utilities/FormInputValidation.java

```diff
@@ -199,6 +199,41 @@ public static boolean assetFormInputValidation(AnchorPane textInputControlAnchor
         return assetValidForm;
     }
 
+    /**
+     * This function validates the user input for asset input forms when editing an asset.
+     * Whenever an input is invalid, it creates and displays an error message for the specific invalid input field.
+     *
+     * @param textInputControlAnchorPane The Anchor pane of the input form
+     * @param assetDescriptionInput      TextInputControl for the asset description
+     * @param manufacturerInput          TextInputControl for the asset manufacturer
+     * @param categoryInput              TextInputControl for the asset category
+     * @param siteInput                  TextInputControl for the asset site
+     * @param locationInput              TextInputControl for the asset location
+     * @author Paul
+     */
+    public static boolean assetEditFormInputValidation(AnchorPane textInputControlAnchorPane, TextArea assetDescriptionInput, TextField manufacturerInput, TextField categoryInput, TextField siteInput, TextField locationInput) {
+        String regexWordAndHyphen = "(?=\\S*[-])([a-zA-Z0-9-]*)|([a-zA-Z0-9]*)"; //Any word containing letters, numbers and hyphens
+        String regexLettersAndHyphen = "(?=\\S*[-])([a-zA-Z-]*)|([a-zA-Z]*)"; //Any word containing letters and hyphens
+        assetValidForm = true;
+        ArrayList<TextInputControl> textInputControls = new ArrayList<>();
+
+        // Initializing the form input validation
+        initializeFormInputValidation(textInputControlAnchorPane, textInputControls);
+
+        logger.info("Start - formInputValidation() -> The form is : {}", assetValidForm);
+
+        // Input validation on the different asset input fields
+        assetDescriptionValidation(textInputControlAnchorPane, assetDescriptionInput, textInputControls.indexOf(assetDescriptionInput));
```

Comment:
There is no validation on the name even though you can edit the asset name. This causes some issues when you enter a name that is too long. And the user can't know because he is brought back to the assets thumbnail screen. He will only notice if he comes back on the asset info page.
![asset_name_1](https://user-images.githubusercontent.com/47158909/113662148-82849880-9675-11eb-83c5-3d88e0cd20e6.PNG)

Reply:
Good catch